waiter/src/waiter/scheduler/kubernetes.clj

```diff
@@ -712,7 +712,8 @@
        :spec {:replicas min-instances
               :selector {:matchLabels {:app k8s-name
                                        :waiter-cluster cluster-name}}
-              :template {:metadata {:annotations {:waiter/port-count (str ports)
+              :template {:metadata {:annotations {:waiter/health-check-port-index (str health-check-port-index)
```

Comment:
Why does this need to be an annotation?

Reply:
The `pod->ServiceInstance` function converts a Kubernetes Pod to a waiter.scheduler/ServiceInstance. Since that function takes no other helpers, we populate the pod with enough metadata to be able to create the instance.

Reply:
Rewrote the code so that service instance does not need a new field.